### PR TITLE
fix(masthead): ssr fix for scroll check

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

#3661 

### Description

An SSR bug was picked up after a fix for #3661, this is adding a check.

### Changelog

**Changed**

- Adding check for `document.body`

### CODE REVIEW

File: packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js [hunk 1]
```diff
@@ -140,15 +140,9 @@ class HeaderMenu extends React.Component {
   handleOnBlur = event => {
     if (!event.currentTarget.contains(event.relatedTarget)) {
       this.setState({ expanded: false, selectedIndex: null });
-      root.document.body.classList.remove(`${prefix}--body__lock-scroll`);
+      root.document?.body?.classList.remove(`${prefix}--body__lock-scroll`);
     }
 
-    const megamenuItems = [
```
Comment (jeffchew) [inline packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js:146]: FYI, I noticed that this was not being used at all so I removed this @annawen1

Comment (asudoh) [reply packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js:146]: I see, a side note @jeffchew what do you think of enabling [ESLint `no-unused-vars` rule](https://eslint.org/docs/rules/no-unused-vars)?

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jeffchew!

File: packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js (hunk 1 above)
Comment (annawen1) [reply packages/react/src/components/carbon-components-react/UIShell/HeaderMenu.js:146]: yep, thanks for catching that!

**Comment by jeffchew** [comment]
@asudoh agreed on the ESLint rule update, I'm actually surprised it's not enabled (thought it was)...

**Pull request merged**